tika-translate/src/main/java/org/apache/tika/language/translate/Lingo24Translator.java

```diff
@@ -43,72 +46,78 @@
-            throw new TikaException(jsonResp.findValue("errors").get(0).asText());
-        }
-    }
-
-    @Override
-    public String translate(String text, String targetLanguage)
-            throws TikaException, IOException {
-        if (!this.isAvailable)
-            return text;
-        LanguageIdentifier language = new LanguageIdentifier(
-                new LanguageProfile(text));
-        String sourceLanguage = language.getLanguage();
-        return translate(text, sourceLanguage, targetLanguage);
-    }
-
-    @Override
-    public boolean isAvailable() {
-        return this.isAvailable;
-    }
+	private static final String LINGO24_TRANSLATE_URL_BASE = "https://api.lingo24.com/mt/v1/translate";
+
+	private static final String DEFAULT_KEY = "dummy-key";
+
+	private String userKey;
+
+	public Lingo24Translator() {
+		Properties config = new Properties();
+		try {
+			config.load(Lingo24Translator.class
+					.getResourceAsStream(
+							"translator.lingo24.properties"));
+			this.userKey = config.getProperty("translator.user-key");
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	@Override
+	public String translate(String text, String sourceLanguage,
+			String targetLanguage) throws TikaException, IOException {
+		if (!this.isAvailable()) {
+			return text;
+		}
+		WebClient client = WebClient.create(LINGO24_TRANSLATE_URL_BASE);
+
+		Response response = client.accept(MediaType.APPLICATION_JSON)
+				.query("user_key", this.userKey).query("source", sourceLanguage)
+				.query("target", targetLanguage).query("q", text).get();
+		BufferedReader reader = new BufferedReader(new InputStreamReader(
+				(InputStream) response.getEntity(), UTF_8));
+		String line = null;
+		StringBuffer responseText = new StringBuffer();
+		while ((line = reader.readLine()) != null) {
+			responseText.append(line);
+		}
+
+		try {
+			ObjectMapper mapper = new ObjectMapper();
+			JsonNode jsonResp = mapper.readTree(responseText.toString());
+			if (jsonResp.findValuesAsText("errors").isEmpty()) {
+				return jsonResp.findValuesAsText("translation").get(0);
+			} else {
+				throw new TikaException(jsonResp.findValue("errors").get(0).asText());
+			}
+		} catch (JsonParseException e) {
+			throw new TikaException("Error requesting translation from '" + sourceLanguage + "' to '" + targetLanguage + "', JSON response from Lingo24 is not well formatted: " + responseText.toString());
+		}
+	}
+
+	public String getUserKey() {
+		return userKey;
+	}
+
+	public void setUserKey(String userKey) {
+		this.userKey = userKey;
+	}
+
+	@Override
+	public String translate(String text, String targetLanguage)
+			throws TikaException, IOException {
+		if (!this.isAvailable())
+			return text;
+		LanguageIdentifier language = new LanguageIdentifier(
+				new LanguageProfile(text));
+		String sourceLanguage = language.getLanguage();
+		return translate(text, sourceLanguage, targetLanguage);
+	}
+
+	@Override
+	public boolean isAvailable() {
```

Comment:
"isAvailable" became a method so it can check for correct setup at any time, which allows for setting the User-Key, before it was expecting it from the properties file only at instantiation time.